reference/language-concepts/namespaces.markdown

```diff
@@ -6,104 +6,76 @@ sorting: 100
 
-```cf3
-    methods:
+## Methods|usebundle
 
-      "namespace demo" usebundle => myspace:mymethod("arg1");
-      "namespace demo" usebundle => mymethod("arg1","arg2");
-```
+Methods promises assume you are referring to a bundle in the same namespace as
+the promiser. To refer to a bundle in another namespace you *must* specify the
+namespace by prefixing the bundle name with the namespace followed by a colon
+(`:`).
 
-To distinguish a body from one in another namespace, you can prefix the body name with the namespace, separated by a colon.
+[%CFEngine_include_example(namespace_methods-usebundle.cf)%]
 
-```cf3
-    files:
-       "/file"
-          create => "true",
-           perms => name1:settings;
-```
+## Bodies
 
-If you don't make any namespace declarations, you'll be in the
-`default` namespace.  Bundles, bodies, classes, and variables from the
-`default` namespace can be accessed like any other:
+Bodies are assumed to be within the same namespace as the promiser. To use a body from another namespace the namespace must be specified by prefixing the body name with the namespace followed by a colon (`:`).
 
-```cf3
-    files:
-      "/file"
-         create => "true",
-          perms => default:settings;
-```
+A common mistake is forgetting to specify `default:` when using bodies from the standard library which resides in the `default` namespace.
 
-If you use the standard library from your own namespace, remember to
-specify this `default:` prefix.
+[%CFEngine_include_example(namespace_bodies.cf)%]
 
-To access classes, variables, or meta-data in bundles in a different namespace, use the
-colon as a namespace prefix:
+## Variables
 
-    $(namespace:bundle.variable)
-    $(namespace:bundle_meta.variable)
+Variables (except for Special Variables) are assumed to be within the same scope
+as the promiser but can also be referenced fully qualified with the namespace.
 
-**Note** that this means that if you are in a namespace that's not `default`, you *must* qualify classes from `default` fully:
+[%CFEngine_include_example(namespace_variable_references.cf)%]
 
-    default:myclass::
-    "do something" if => "default:myotherclass";
+[Special variables][Special Variables] are always accessible without a namespace
+  prefix. For example, `this`, `mon`, `sys`, and `const` fall in this category.
 
-### Namespacing of classes and variables created in policy
+[%CFEngine_include_example(namespace_special_var_exception.cf)%]
 
-In policy, you can't create classes outside your own namespace.  So
-the following, for example, will create the class `mynamespace:done`
-if it runs in the namespace `mynamespace`.
+**Notes:**
 
-```cf3
-    files:
-      "/file"
-         create => "true",
-         action => if_repaired("done");
-```
+- The [`variables` Augments key][Augments#variables] defines variables in the
+  `main` bundle of the `data` namespace by default but supports seeding variable
+  values in any specified namespace.
```

Comment:
an example would be nice just to be clear: aka variables item foo would be referred to as data:main.foo right?

Reply:
Yeah, but I was also think to just convert this into a see also reference. There's a relationship I want to point out but I don't want to distract from what this is about.